mono/mini/debugger-agent.c

```diff
@@ -3899,7 +4041,9 @@ thread_end (MonoProfiler *prof, uintptr_t tid)
 			/* FIXME: Maybe we need to free this instead, but some code can't handle that */
 			tls->terminated = TRUE;
 			/* Can't remove from tid_to_thread, as that would defeat the check in thread_start () */
+#ifndef IL2CPP_MONO_DEBUGGER
 			MONO_GC_UNREGISTER_ROOT (tls->thread);
```

Comment:
Not exactly sure how things work, but why don't we need to register with the GC for the il2cpp debugger?